File: OOP/abstract.cpp


#include<bits/stdc++.h>
using namespace std;


class Shape{
        public:
        virtual void sides() = 0;
};


class Rectangular : public Shape{
        public:
        void sides(){
                cout << "My opposite sides are equal" << endl;
        }
};




int main(){
        Rectangular rectangle;
        rectangle.sides();

        return 0;
}

File: OOP/abstract.py
from abc import ABC, abstractmethod

class Shape(ABC):
        @abstractmethod
        def sides(self):
                pass


class Rectangular(Shape):
        def sides(self):
                print("My opposite sides are equal")

class Tringle(Shape):
        def sides(self):
                print("My sides are can be equal or not")


triangle = Tringle()
triangle.sides()

rectangle  = Rectangular()
rectangle.sides()